contracts/chain-adapters/Arbitrum_Adapter.sol

```diff
@@ -168,6 +225,13 @@ contract Arbitrum_Adapter is AdapterInterface, CircleCCTPAdapter {
 
     ArbitrumL1ERC20GatewayLike public immutable L1_ERC20_GATEWAY_ROUTER;
 
+    // This token can be used to pay for l1 to l2 messages if its configured by an Arbitrum orbit chain.
+    // If this is initialized to be non-zero, then this token will be pulled from the customL2GasTokenFunder contract
+    // to pay for l1 to l2 messages.
+    IERC20 public immutable CUSTOM_GAS_TOKEN;
+
+    address public immutable CUSTOM_GAS_TOKEN_FUNDER;
```

Comment:
Will this contract be added in a separate PR?

Reply:
yeah let's just assume for now it is some contract that has granted approval to this contract to pull all of its funds. WDYT?

Reply:
Do you think we need to audit that contract?

Reply:
probably but i didn't want to build it in this PR. It could even be an EOA that we seed with funds, right?

Reply:
i think the key thing to center around is whether we should "pull" from the funder via transferFrom or should there be an interface that exposes a function like `pull()` that we can cull to get funds

Reply:
I guess one problem with just calling transferFrom from a set address is that LP's might worry that the address gets set to their EOA, which has approved the HubPool to spend their tokens, and can be used as a vector to take their funds.

Reply:
wdyt of this simple funder: edc3fd86b226a7b6d5321f9853c91f28d5a9c8f7

Reply:
Yeah, I like that we're not just doing a transferFrom from an arbitrary address. +1 for simple funder contract.